lib/rules/button-has-type.js

```diff
@@ -149,9 +150,11 @@ module.exports = {
         }
 
         const props = node.arguments[1].properties;
-        const typeProp = props.find((prop) => prop.key && prop.key.name === 'type');
+        const typeProp = props.find(
+          (prop) => prop.type === "Property" && prop.key && prop.key.type === "PrivateIdentifier" && prop.key.name === 'type'
+        );
 
-        if (!typeProp) {
+        if (typeProp.type !== "Property") {
```

Comment:
these changes aren't simple annotation changes. can they be extracted at least to a different commit, if not to a different PR?

Reply:
I see. My apologies 🙇‍♂️. 
Would it be acceptable to create a PR that adds annotations only to rules where type changes are unnecessary?
Or would it be better to add annotations to all rules listed below, and if changes are needed, ignore the entire file (@ts-nocheck)?

Reply:
yes, that would be fine.

any non-type changes that are needed must be accompanied by tests - if the ostensible type error can't be reproduced, then the type system is wrong.